arroyo/processing/strategies/produce.py

```diff
@@ -71,9 +73,20 @@ def poll(self) -> None:
                 if not future.done():
                     break
 
+                try:
+                    result = future.result()
+                    self.__metrics.increment(
+                        "arroyo.producer.produce_status", tags={"status": "success"}
+                    )
+                except Exception as e:
+                    self.__metrics.increment(
+                        "arroyo.producer.produce_status", tags={"status": "error"}
+                    )
+                    raise e
```

Comment:
We need to identify the error code and add it as a tag. Not all errors should make us burn our SLO. For example messages that are too large would still show up here. Those should not make us burn the SLO.